src/proficiency.cpp

```diff
@@ -383,19 +383,22 @@ void book_proficiency_bonus::deserialize( JsonIn &jsin )
     optional( jo, was_loaded, "fail_factor", fail_factor, default_fail_factor );
     optional( jo, was_loaded, "time_factor", time_factor, default_time_factor );
     optional( jo, was_loaded, "include_prereqs", include_prereqs, default_include_prereqs );
-}
-
-book_proficiency_bonus &book_proficiency_bonus::operator+=( const book_proficiency_bonus &rhs )
-{
-    if( !id.is_empty() && id != rhs.id ) {
-        debugmsg( "ERROR: Tried to add two book proficiency bonuses with different ids" );
-        // can't add them together unless the ids are the same
-        return *this;
+    if( fail_factor < 0 ) {
+        debugmsg( "fail_factor for proficiency %s must not be <0, value: %f", id.str(), fail_factor );
+        fail_factor = 0;
+    }
+    if( fail_factor >= 1 ) {
+        debugmsg( "fail_factor for proficiency %s must not be >=1, value: %f", id.str(), fail_factor );
+        fail_factor = 0.999f;
+    }
+    if( time_factor < 0 ) {
+        debugmsg( "time_factor for proficiency %s must not be <0, value: %f", id.str(), time_factor );
+        time_factor = 0;
+    }
+    if( time_factor >= 1 ) {
+        debugmsg( "time_factor for proficiency %s must not be >=1, value: %f", id.str(), time_factor );
+        time_factor = 0.999f;
     }
```

Comment:
Consider using `jo.throw_error` for these instead of `debugmsg` so that they provide context as to where the error is.  (The disadvantage is that they will then become fatal errors and prevent the game starting, but I don't think that's too big a deal so long as we don't know of any examples that would trigger these errors).